src/rrd_dump.c

```diff
@@ -569,6 +569,7 @@ int rrd_dump(
                           "[--no-header|-n]\n"
                           "[--daemon|-d address]\n"
                           "file.rrd [file.xml]", options.argv[0]);
+            if (opt_daemon) free(opt_daemon);
```

Comment:
Personally no a fan of this style of shorthand but not sure what the project rules are about it. That and it should be a NULL test.
```c
if (opt_daemon != NULL)
   free(opt_daemon);
```